src/main/java/org/snomed/snowstorm/core/data/services/AutoMerger.java

```diff
@@ -0,0 +1,518 @@
+package org.snomed.snowstorm.core.data.services;
+
+import org.snomed.snowstorm.core.data.domain.*;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+@Component
+public class AutoMerger {
+    private enum ComponentType {
+        Concept, Description, Relationship,
+        ClassAxiom, GciAxiom, LanguageReferenceSetMember
+    }
+
+    /**
+     * Construct a new instance of Concept which is a combination of both sourceConcept and targetConceptNew. The new, auto-merged
+     * instance will be based on sourceConcept with edits belonging targetConceptNew applied on top. To find the edits belonging
+     * to targetConceptNew, the instance will be compared against targetConceptOld. If any parameter given is null,
+     * then a new, blank instance will be returned.
+     *
+     * @param sourceConcept    The state that will be the baseline for the new, auto-merged Concept.
+     * @param targetConceptOld The state to help identify which edits belong to targetConceptNew.
+     * @param targetConceptNew Edits belonging to this Concept will be re-applied to the new, auto-merged Concept.
+     * @return A new instance of Concept which is a combination of both sourceConcept and targetConceptNew.
+     */
+    public Concept autoMerge(Concept sourceConcept, Concept targetConceptOld, Concept targetConceptNew) {
+        if (sourceConcept == null || targetConceptOld == null || targetConceptNew == null) {
+            return new Concept();
+        }
```

Comment:
I have replaced the existing "automerge" function with this class. Having the logic in its own class let me write more tests. All current tests pass with this new (moderately the same) implementation.